### ISSUE TITLE
Django-3.1 native async views hang on internal requests

### ISSUE THREAD

**Issue by imbolc** [django/channels#1491]
Django native async views can handle multiple concurrent requests to another internal pages.
But wrapped in channels application it hangs on a low level of concurrency.
It doesn't happen with external requests though.

I published an example as a gist: https://gist.github.com/imbolc/b7c7192aba56ba0d8beb7b2a7b12928c

**Comment by carltongibson** [comment]
Thanks for the report. Needs a bit of investigating to see what's going on there... 🤔

**Comment by caesarleong** [comment]
I do facing this case too. Hope it can solve soon.
@imbolc do you have any workaround can bypass this problem now? Thanks

**Comment by imbolc** [comment]
@caesarleong you can split your app by nginx so http part won't be wrapped into channels like: 
```
/ws  --> channels.asgi
/api --> django.asgi
```

**Comment by jamiegau** [comment]
Yes, I needed to update a web app, and decided to pull it up to current release for all libs involved (As of today).  And it went well but for...
When `channels` is in `INSTALLED_APPS`,  when I run the test server `mamange.py runserver` and hit the server with concurrent requests, I will then see stall/hang requests and the browser will report them pending forever.

In trying to sort this out, I ended up running gunicorn instead..  I needed to set (on a Mac) "`OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES`" env variables to get that not to loop on failed fork(), but it did appear to work.    `manage.py runserver` was not affected by setting that env-variable.

I hope this info is useful,  I am keen to keep moving forward and not having to roll back.
If any other tests are suggested and I have some time.  I'll have a look.  This is a show stopper to channels, so I imagine its going to get attention.
